pkg/syncer/metadatasyncer.go

```diff
@@ -855,6 +857,89 @@ func addLabelsToTopologyVCMap(ctx context.Context, nodeTopoObj csinodetopologyv1
+	)
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(
+		obj.(*unstructured.Unstructured).Object, &cnsvolumeoperationrequestObj)
+	if err != nil {
+		log.Errorf("cnsvolumeoperationrequestCRAdded: failed to cast object %+v to %s. Error: %+v", obj,
+			cnsvolumeoperationrequest.CRDSingular, err)
+		return
+	}
+	// Check for the below set of conditions:
+	// 1. Cnsvolumeoperationrequest object's StorageQuotaDetails should not be nil
+	// 2. Cnsvolumeoperationrequest object's StorageQuotaDetails.Reserved should not be nil
+	if cnsvolumeoperationrequestObj.Status.StorageQuotaDetails != nil &&
+		cnsvolumeoperationrequestObj.Status.StorageQuotaDetails.Reserved != nil {
+		restConfig, err := config.GetConfig()
+		if err != nil {
+			log.Errorf("cnsvolumeoperationrequestCRAdded: failed to get Kubernetes config. Err: %+v", err)
+			return
+		}
+		cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, cnsoperatorv1alpha1.GroupName)
+		if err != nil {
+			log.Errorf("cnsvolumeoperationrequestCRAdded: Failed to create CnsOperator client. Err: %+v", err)
+			return
+		}
+		namespace := cnsvolumeoperationrequestObj.Status.StorageQuotaDetails.Namespace
+		storagePolicyUsageList := &storagepolicyusagev1alpha1.StoragePolicyUsageList{}
+		err = cnsOperatorClient.List(ctx, storagePolicyUsageList, client.InNamespace(namespace))
+		if err != nil {
+			log.Errorf("failed to list %s CR from supervisor namespace %q. Error: %+v",
+				storagepolicyusagev1alpha1.CRDSingular, namespace, err)
+			return
+		}
+		foundStoragePolicyUsageCR := false
+		oldStoragePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+		newStoragePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+		if len(storagePolicyUsageList.Items) > 0 {
+			for _, item := range storagePolicyUsageList.Items {
+				policyId := cnsvolumeoperationrequestObj.Status.StorageQuotaDetails.StoragePolicyId
+				scName := cnsvolumeoperationrequestObj.Status.StorageQuotaDetails.StorageClassName
+				if item.Spec.StoragePolicyId == policyId && item.Spec.StorageClassName == scName {
+					log.Debugf("cnsvolumeoperationrequestCRAdded: Found storagePolicyUsage CR with matching "+
+						"storagePolicyId: %q & storageClassName: %q in namespace: %q", policyId, scName,
+						namespace)
+					oldStoragePolicyUsageCR = item.DeepCopy()
+					newStoragePolicyUsageCR = item.DeepCopy()
+					foundStoragePolicyUsageCR = true
+					break
+				}
+			}
+		}
+		if !foundStoragePolicyUsageCR {
+			log.Errorf("unable to find matching %q CR with PolicyId: %q & PolicyName: %q from "+
+				"supervisor namespace %q", storagepolicyusagev1alpha1.CRDSingular,
+				cnsvolumeoperationrequestObj.Status.StorageQuotaDetails.StoragePolicyId,
+				cnsvolumeoperationrequestObj.Status.StorageQuotaDetails.StorageClassName,
+				namespace)
+			return
+		}
+		if newStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage != nil {
```

Comment:
following previous comment we can check for `storagePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage != nil {`

Reply:
Done